### ISSUE TITLE
Brave browser sync is being disconnected frequently or a few minutes after being set-up.

### ISSUE THREAD

**Issue by SS1113** [brave/browser-android-tabs#1535]
## Description  
Brave browser sync is being disconnected frequently or a few minutes after being set-up

## Steps to reproduce 
   1. Setup sync 
   2. Check after few minutes it'll be disconnected, I have to reconnect it.

## Expected result

Sync should not be disconnected

## Issue reproduces how often 
Everytime (100% reproducible everytime)

## Issue happens on 
   - Current Play Store version? - Yes. 
   - Beta build? - Didn't check.

## Device details
   - Install type (ARM, x86): ARM
   - Device (Phone, Tablet, Phablet): Phone Xiaomi Mi A1
   - Android version: Android 9 Pie with April security patch
   
## Brave version
1.0.92

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? - No
- Is the issue reproducible on the latest version of Chrome? Chrome doesn't have Brave sync so I couldn't test

### Additional information
 No

**Comment by SS1113** [comment]
I clear browsing data like cookies, saved and everything except cache. Could that be the reason why the Sync setup is being cleared from my device?

**Comment by Kshitij-agrawal2010** [comment]
This happens with me too. It disconnects every 2-3 days.

**Comment by samartnik** [comment]
Steps to reproduce:
- Connect to or create Sync chain.
- Close Brave.
- Disconnect from network.
- Run Brave and go to Sync menu.

**Comment by btlechowski** [comment]
Verification passed with BraveModernarm v1.0.98-rc.1 on Samsung s7. I verified the steps from https://github.com/brave/browser-android-tabs/issues/1535#issuecomment-501858870. Tested different scenarios when going offline(I noticed that it takes longer then usual to sync after going back online, usually restarting the browser or opening a page speeds up the process). Also tested sync scenarios from manual testrun.
